pydra/engine/core.py

```diff
@@ -225,13 +225,19 @@ def checksum(self):
             self.inputs._graph_checksums = [nd.checksum for nd in self.graph_sorted]
 
         input_hash = self.inputs.hash
-        if self.state is None:
+        if self.state is None and not is_workflow(self):
             self._checksum = create_checksum(self.__class__.__name__, input_hash)
         else:
-            # including splitter in the hash
-            splitter_hash = hash_function(self.state.splitter)
+            hash_list = [input_hash]
+            if self.state:
```

Comment:
i'm not sure if state should matter? we don't save any working directories for workflows with state right, a workflow with state first gets converted to without state and then run.

Reply:
i don't have a good answer... was also wondering why i have this if statement here, and not in `checksum_sates` and even created #259 to not forget that i have to review these two functions. 
let me just do it within this pr, I will review the issue, I know that removing this part leads to at least one failure in our tests

Reply:
it has to be, since you can have two workflows that have the same inputs but different splitters  (e.g., scalar vs outer)

Reply:
but a workflow with a state is never run or cached. however, i think i know why this is needed, to compute the checksum of a nested workflow with state. so yes to compute the workflow graph checksum this is required.

Reply:
correct, this is used for checking the nodes within wf, not to create directory